src/transformers/hf_argparser.py

```diff
@@ -103,7 +103,7 @@ def _parse_dataclass_field(parser: ArgumentParser, field: dataclasses.Field):
                 kwargs["default"] = field.default
             else:
                 kwargs["required"] = True
-        elif field.type is bool or field.type is Optional[bool]:
+        elif field.type is bool or field.type == Optional[bool]:
```

Comment:
I'm wondering whether we should use "==" everywhere for consistency, as we now have both "is" and "==".

Reply:
Fine by me. You should also fix the test on line 143 of this file :-)